packages/venia-concept/src/components/ProductQuantity/quantity.js

```diff
@@ -1,41 +1,86 @@
-                value: number
-            })
-        )
+const Quantity = props => {
+    const { classes, initialValue, onValueChange, formApi, formState } = props;
+
+    const incrementQty = formApi => {
+        var currentQuantity = parseInt(formApi.getValue('quantity')) || 0;
+        const incrementedQty = currentQuantity + 1;
+        formApi.setValue('quantity', incrementedQty);
+        onValueChange(formApi.getValue('quantity'));
     };
 
-    static defaultProps = {
-        selectLabel: "product's quantity"
+    const decrementQty = formApi => {
+        var currentQuantity = parseInt(formApi.getValue('quantity')) || 0;
+        const decrementedQty = currentQuantity - 1;
+        formApi.setValue('quantity', decrementedQty);
+        onValueChange(formApi.getValue('quantity'));
     };
 
-    render() {
-        const { classes, selectLabel, ...restProps } = this.props;
+    const validateQuantity = value => {
+        var validateRegex = /^[1-9]\d*(\.\d+)?$/;
+        return value <= 0
+            ? 'Please enter a quantity greater than 0.'
+            : value == undefined || !validateRegex.test(value)
+            ? 'Enter a quantity of 1 or more.'
+            : undefined;
+    };
 
-        return (
-            <div className={classes.root}>
-                <Select
-                    {...restProps}
+    return (
+        <div className={classes.quantityFieldbox}>
+            <label htmlFor="quantity">
+                <Text
+                    initialValue={initialValue}
+                    onValueChange={onValueChange}
+                    id="quantity"
+                    type="number"
+                    min="1"
                     field="quantity"
-                    aria-label={selectLabel}
-                    items={mockData}
+                    validateOnChange
+                    validate={validateQuantity}
+                    className={classes.quantityField}
                 />
-            </div>
-        );
-    }
-}
+                <button
+                    disabled={formApi.getValue('quantity') <= 1 ? true : false}
+                    aria-label="Quantity Decrement"
+                    type="button"
+                    onClick={() => {
+                        decrementQty(formApi);
+                    }}
+                >
+                    <Minus />
+                </button>
+                <button
+                    aria-label="Quantity Increment"
+                    type="button"
+                    onClick={() => {
+                        incrementQty(formApi);
+                    }}
+                >
+                    <Plus />
+                </button>
+            </label>
+            <p className={classes.errors}>{formState.errors.quantity}</p>
+        </div>
+    );
+};
+
+Quantity.propTypes = {
+    classes: shape({
+        root: string,
+        quantityField: string
+    }),
+    items: arrayOf(
+        shape({
+            value: number
+        })
+    )
```

Comment:
Can you add the `onValueChange` prop here? It is type `func`, and it is required.
```suggestion
    ),
    onValueChange: func.isRequired
```